### ISSUE TITLE
Collapse by Conditioning: Training Class-conditional GANs with Limited Data

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#152]
## TL; DR

- Gradually shift from unconditional GAN training to conditional training (linearly) for diverse image generation with limited training set.

## Motivation

- Conditional GANs suffer more from mode collapse compared to unconditional GANs.
- Mode collapse seems to happen early in the training stage, hence move from unconditional to conditional using simple linear scaling.

![image](https://user-images.githubusercontent.com/46308407/152489667-c35c243a-b983-474e-a07c-9947963d2b9e.png)
![image](https://user-images.githubusercontent.com/46308407/152489700-f6e22daa-f5d6-402b-af7a-b91c0291f4d0.png)

### Data Augmentations
- Adaptive Data Augmentation (ADA, NIPS'20)
- Differentiable Augmentation (DiffAug, NIPS'20)

### Regularization
- Lecam-GAN (CVPR'21)

## Method

![image](https://user-images.githubusercontent.com/46308407/152489572-bba39262-d9b3-444c-a9dd-ec7922dd3331.png)

- Implement on Conditional StyleGAN2 (class info provided through style mapping network)
- Set additional parameter lambda_t, which is linearly scaled up from 0 to 1 during the transition phase.

![image](https://user-images.githubusercontent.com/46308407/152489767-045a9bcc-02fb-4821-a24e-81da8b98cc80.png)

- Set T_s=2000, T_e=4000 (start and end of transition phase, respectively)

## Experiments

![image](https://user-images.githubusercontent.com/46308407/152489813-14291bb0-b458-401b-99fa-371a86d8035a.png)
![image](https://user-images.githubusercontent.com/46308407/152489937-f01903bd-3818-4238-86d3-ee54718964ca.png)
![image](https://user-images.githubusercontent.com/46308407/152489953-00b53198-3073-4ab2-85fb-cf7e1f730c34.png)
![image](https://user-images.githubusercontent.com/46308407/152490014-d63401d5-5bdf-4af9-9769-beb877df7457.png)
![image](https://user-images.githubusercontent.com/46308407/152490036-df805d46-dac6-49ae-9270-72208b7cbccb.png)

## Comments

- Very simple method but made powerful by persuasive writing. (Good paper)
- Interesting observation + problem identification (stronger mode collapse in conditional GANs under limited data)
- Note that *the images were processed by the authors to only contain the faces of the animals* (ImageNet-Carnivores)
- Alignment is very important, even with relatively abundant data (2k~5K)
- Need to define problem as well as target dataset in *cleverly*.
